adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -196,39 +197,66 @@ func (b *BM25Searcher) wand(
 	averagePropLength = averagePropLength / float64(len(properties))
 
 	// preallocate the results (+1 is for full query)
-	results := make(terms, 0, len(queryTextTerms)+len(queryStringTerms)+1)
-	indices := make([]map[uint64]int, 0, len(queryTextTerms)+len(queryStringTerms)+1)
+	fullQueryLength := 0
+	if len(propertyNamesFullQuery) > 0 {
+		fullQueryLength = 1
+	}
+	stringLength := 0
+	if len(propertyNamesString) > 0 {
+		stringLength = len(queryStringTerms)
+	}
+	textLength := 0
+	if len(propertyNamesText) > 0 {
+		textLength = len(queryTextTerms)
+	}
+	results := make(terms, textLength+stringLength+fullQueryLength)
```

Comment:
`textLength+stringLength+fullQueryLength` is used a few times, could probably be a variable

Reply:
done